### PR TITLE
Set up jekyll-apple-help build pipeline

### PR DESCRIPTION
**Pull request by TzviPM**
## Summary
Generate pages for help book from markdown and JSON content using https://github.com/chuckhoupt/jekyll-apple-help. This will make it easier for newcomers to contribute help content to the project with confidence.

### High-level overview

The XCode project now has a folder called `XiEditorHelp`, which contains the following structure:
screen shot 2019-02-21 at 1 16 18 pm

This folder gets compiled as part of the XiEditorHelp build target, which generates a `.help` bundle:
screen shot 2019-02-21 at 1 16 26 pm

The main files we'll want to edit for creating content are in the `_English.lproj` folder nested under `XiEditorHelp`. This folder contains numerous `.md` files with `yml`-formatted front matter.

Front matter should contain a `title` attribute as well as a `description`, and optionally an `order` to establish a consistent ordering of help topics.

An additional `group` property can specify a title for a folder in which to group topics together, aggregated by this given title.

the `help` bundle will be automatically rebuilt as part of the application build step.

---

Supersedes https://github.com/xi-editor/xi-mac/pull/400
Part of #252 

## Checklist
Example:

- [x] Set up jekyll-apple-help
- [x] Create content in md format
- [x] Keyboard shortcuts stuff

## Review Checklist

- [x] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-mac/master.

### CODE REVIEW

**Comment by cmyr** [comment]
@tzvipm Cool! Could you provide a bit of an overview of how this is intended to work, when it should be run, etcetera? Just a high level walk through, so I have a better sense of what to look for in review.

**Comment by tzvipm** [comment]
@cmyr I updated the description just now; is this kinda what you were looking for in a high-level overview?

File: XiEditorHelp/_config.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Site settings
```
Comment (tzvipm) [inline XiEditorHelp/_config.yml]: copied from JekyllHelp

File: XiEditorHelp/_includes/footer.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{% comment %}
```
Comment (tzvipm) [inline XiEditorHelp/_includes/footer.html]: copied from JekyllHelp

File: XiEditorHelp/_includes/head.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<head>
```
Comment (tzvipm) [inline XiEditorHelp/_includes/head.html]: copied from JekyllHelp

File: XiEditorHelp/_includes/header.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<header class="banner">
```
Comment (tzvipm) [inline XiEditorHelp/_includes/header.html]: copied from JekyllHelp

File: XiEditorHelp/_layouts/default.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tzvipm) [inline XiEditorHelp/_layouts/default.html]: copied from JekyllHelp

File: XiEditorHelp/_layouts/help-page.html [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (tzvipm) [inline XiEditorHelp/_layouts/help-page.html]: copied from JekyllHelp

File: XiEditorHelp/_layouts/title-page.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (tzvipm) [inline XiEditorHelp/_layouts/title-page.html]: copied from JekyllHelp

File: XiEditorHelp/_plugins/help-filters.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
+module Jekyll
```
Comment (tzvipm) [inline XiEditorHelp/_plugins/help-filters.rb]: copied from JekyllHelp

File: XiEditorHelp/_plugins/page-rel-root.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# Generator to add relative path to root to every page/collection doc with the key rel_root
```
Comment (tzvipm) [inline XiEditorHelp/_plugins/page-rel-root.rb]: copied from JekyllHelp

File: XiEditorHelp/_sass/_apple-help.scss [hunk 10]
```diff
@@ -0,0 +1,223 @@
+
```
Comment (tzvipm) [inline XiEditorHelp/_sass/_apple-help.scss]: copied from JekyllHelp

File: XiEditorHelp/_sass/_apple-help.scss [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+- Delete the banner at html > body > div > header
+
+Now the page looks/works the same as in HelpViewer (except for responsive styles).
+
+*/
+
+// Base variable defaults, which can be overridden
+
+// $base-font-family: HelveticaNeue, 'Helvetica Neue', Helvetica, sans-serif !default;;
+$base-font-family: 'Roboto', 'Helvetica', sans-serif !default;
```
Comment (tzvipm) [inline XiEditorHelp/_sass/_apple-help.scss]: changed this line to match the fonts in xi-editor.io

File: XiEditorHelp/_sass/_base.scss [hunk 12]
```diff
@@ -0,0 +1,181 @@
+/**
```
Comment (tzvipm) [inline XiEditorHelp/_sass/_base.scss]: copied from JekyllHelp

File: XiEditorHelp/_sass/_syntax-highlighting.scss [hunk 13]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (tzvipm) [inline XiEditorHelp/_sass/_syntax-highlighting.scss]: copied from JekyllHelp

File: XiEditorHelp/css/main.scss [hunk 14]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (tzvipm) [inline XiEditorHelp/css/main.scss]: copied from JekyllHelp

File: XiEditorHelp/jekyll-server.command [hunk 15]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (tzvipm) [inline XiEditorHelp/jekyll-server.command]: copied from JekyllHelp

File: XiEditorHelp/nav.js [hunk 16]
```diff
@@ -0,0 +1,66 @@
+// Nav - simple menu folding in Vanilla JS (http://vanilla-js.com)
```
Comment (tzvipm) [inline XiEditorHelp/nav.js]: copied from JekyllHelp

**Review by TzviPM — COMMENTED** [review]
made some comments to clarify which changes are just copy-paste from jekyll-apple-help.

**Comment by jansol** [comment]
Comments in the copied files linking to the source would be more useful IMHO...

**Comment by nangtrongvuon** [comment]
+1 to @jansol's comment, perhaps it's better to have the comments in the copied files (or just omit them entirely). It's kind of making the PR a bit cluttered and hard to read.

**Review by cmyr — COMMENTED** [review]
Was trying to take another look at this today @tzvipm  (sorry for letting it sit!) but I'm getting an error when I build the project about not finding jekyll?

I do have jekyll installed, but I think Xcode clears `PATH` before running scripts, so I think there will be some other trick needed, here.

I assume this is working for you, do you have any sense of what the problem might be?

**Comment by jansol** [comment]
@tzvipm bump. Would be nice to get this out of the way...

**Pull request closed without merging**